includes/components/class-attribute.php

```diff
@@ -307,6 +310,72 @@ public function register_models() {
 		}
 	}
 
+	/**
+	 * Validates unique model attributes.
+	 *
+	 * @param array  $errors Error messages.
+	 * @param object $model Model object.
+	 * @return array
+	 */
+	public function validate_unique_attributes( $errors, $model ) {
+		if ( empty( $errors ) ) {
+
+			// Set unique fields.
+			$fields = [];
+
+			foreach ( $model->_get_fields() as $field_name => $field ) {
+				if ( ! hp\get_array_value( $field->get_args(), 'unique' ) || ! $field->get_value() ) {
+					continue;
+				}
+
+				$fields[] = [
+					'name'  => $field_name,
+					'value' => $field->get_value(),
+					'label' => $field->get_label(),
+				];
+			}
+
+			// Check fields.
+			if ( ! $fields ) {
+				return $errors;
+			}
+
+			// Get model class.
+			$model_class = hp\create_class_instance( '\HivePress\Models\\' . $model::_get_meta( 'name' ) );
```

Comment:
If I'm not mistaken you can call the query directly on the current model object $model, this will not affect the object itself because query() creates an instance of the model query separate from the model object